jni/src/test/java/com/horizen/certnative/NaiveThresholdSignatureWKeyRotationProofTest.java

```diff
@@ -0,0 +1,276 @@
+        Random r = new Random();
+
+        scId = FieldElement.createRandom();
+        prevScId = FieldElement.createRandom();
+
+        backwardTransferCount = r.nextInt(backwardTransferCount + 1);
+        // Create dummy Backward Transfers
+        List<BackwardTransfer> btList = new ArrayList<>();
+        for(int i = 0; i < backwardTransferCount; i++)
+            btList.add(BackwardTransfer.getRandom(r));
+
+        List<BackwardTransfer> prevBtList = new ArrayList<>();
+        for(int i = 0; i < backwardTransferCount; i++)
+            prevBtList.add(BackwardTransfer.getRandom(r));
+
+        // Compute keys and signatures
+        List<SchnorrKeyPair> signingKeyPairList = new ArrayList<>();
+        List<SchnorrPublicKey> signingPublicKeyList = new ArrayList<>();
+        List<SchnorrPublicKey> masterPublicKeyList = new ArrayList<>();
+        List<SchnorrPublicKey> updatedSigningKeyList = new ArrayList<>();
+        List<SchnorrPublicKey> updatedMasterKeyList = new ArrayList<>();
+        List<SchnorrSignature> updatedSigningKeysSkSignaturesList = new ArrayList<>();
+        List<SchnorrSignature> updatedSigningKeysMkSignaturesList = new ArrayList<>();
+        List<SchnorrSignature> updatedMasterKeysSkSignaturesList = new ArrayList<>();
+        List<SchnorrSignature> updatedMasterKeysMkSignaturesList = new ArrayList<>();
+        List<SchnorrSignature> signatureList = new ArrayList<>();
+
+        for (int i = 0; i<keyCount; i++) {
+            SchnorrKeyPair keyPair = SchnorrKeyPair.generate();
+
+            assertNotNull("Key pair generation was unsuccessful.", keyPair);
+            assertTrue("Public key verification failed.", keyPair.getPublicKey().verifyKey());
+
+            signingKeyPairList.add(keyPair);
+            signingPublicKeyList.add(keyPair.getPublicKey());
+            updatedSigningKeyList.add(keyPair.getPublicKey());
+
+            SchnorrKeyPair masterKeyPair = SchnorrKeyPair.generate();
+
+            assertNotNull("Key pair generation was unsuccessful.", masterKeyPair);
+            assertTrue("Public key verification failed.", masterKeyPair.getPublicKey().verifyKey());
+
+            masterPublicKeyList.add(masterKeyPair.getPublicKey());
+            updatedMasterKeyList.add(masterKeyPair.getPublicKey());
+            masterKeyPair.getSecretKey().freeSecretKey();
+        }
+
+        keysSignaturesList = new ValidatorKeysUpdatesList(
+            signingPublicKeyList,
+            masterPublicKeyList,
+            updatedSigningKeyList,
+            updatedMasterKeyList,
+            updatedSigningKeysSkSignaturesList,
+            updatedSigningKeysMkSignaturesList,
+            updatedMasterKeysSkSignaturesList,
+            updatedMasterKeysMkSignaturesList
+        );
+        validatorsKeysRoot = keysSignaturesList.getKeysRootHash(keyCount);
+        prevValidatorsKeysRoot = keysSignaturesList.getUpdatedKeysRootHash(keyCount);
```

Comment:
The name of this 2 variables seem to be swapped, which seems to lead to misuse of them later on in the code (e.g., when setting custom fields of the withdrawal certificates)